geneate_dynein_conformation_graphs/calculate_stats.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import pandas as pd
+import os 
+from scipy.stats import chi2_contingency
+
+rawdata_folder = 'Data'
+analysis_folder = 'Results'
+results_folder = "Stats"
+
+data = []
+
+# Ensure output folder exists
+if not os.path.exists(f"{analysis_folder}/{results_folder}"):
+    os.makedirs(f"{analysis_folder}/{results_folder}")
+
+for foldername in os.listdir(rawdata_folder):
+    if foldername == 'poly01Tomo01':
+        continue
+
+    input_file = f'{foldername}_bin4_points.csv'
+    input_file_count = f'{foldername}_count.csv'
+    input_file_path = f"{analysis_folder}/{foldername}/{input_file}"
+    input_file_count_path = f"{analysis_folder}/{foldername}/{input_file_count}"
+
+    export_file = f"{foldername}_statistics.txt"
+    export_file_path = f"{analysis_folder}/{results_folder}/{export_file}"
+
+    if os.path.exists(export_file_path):
+        os.remove(export_file_path)
+
+    print(f"Currently gettings statistics for {foldername}")
+
+    for dynein_class in [[1],[2,4],[3]]:
+        # Ensure input file exists
+        if not os.path.exists(input_file_path) and os.path.exists(input_file_count_path):
```

Comment:
this should have parenthesis for the correct check (otherwise `not` only covers the first statement)
```suggestion
        if not (os.path.exists(input_file_path) and os.path.exists(input_file_count_path)):
```